src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -300,8 +311,19 @@ private ReadOnlyMemory<byte> ReadChunkFromConnectionBuffer(int maxBytesToRead, C
                                     _state = ParsingState.Done;
                                     _connection.CompleteResponse();
                                     _connection = null;
+
+                                    // Reached EOF.
+                                    _response.TrailingHeaders = _trailers ??
+                                        HttpResponseHeaders.EmptyResponseHeader;
```

Comment:
What prevents someone from doing `response.TrailingHeaders.TryAddWithoutValidation(name, value)` and modifying the EmptyResponseHeaders collection?

Reply:
Good catch, I should make this immutable.